src/main/java/org/jenkinsci/plugins/workflow/steps/MailStep.java

```diff
@@ -145,10 +146,10 @@ private MimeMessage buildMimeMessage() throws UnsupportedEncodingException, Mess
             if (step.bcc != null) {
                 messageBuilder.addRecipients(step.bcc, Message.RecipientType.BCC);
             }
-            if (step.charset != null) {
+            if (Util.fixEmpty(step.charset) != null) {
```

Comment:
:bug: do not do this here. Rather do it in the setters. Read the [guide](https://github.com/jenkinsci/pipeline-plugin/blob/master/DEVGUIDE.md#handling-default-values).